# This is a patch for gnucobol-svn to update it to gnucobol-svn-patched
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -Naud 'gnucobol-svn/cobc/tree.c' 'gnucobol-svn-patched/cobc/tree.c'
Index: ./cobc/tree.c
--- ./cobc/tree.c	Wed Jul  8 19:14:35 2020
+++ ./cobc/tree.c	Tue Jul 21 08:14:30 2020
@@ -204,8 +204,6 @@
 static	int	prev_expr_warn[EXPR_WARN_PER_LINE] = {0,0,0,0,0,0,0,0};
 static	int	prev_expr_tf[EXPR_WARN_PER_LINE] = {0,0,0,0,0,0,0,0};
 
-static struct cb_report *report_checked = NULL;
-
 /* Local functions */
 
 static int
@@ -3937,8 +3935,8 @@
 	struct cb_reference	*ref;
 	int		k;
 
-	if (report_checked != r) {
-		report_checked = r;
+	if (!r->checked) {
+		r->checked = 1;
 		if (r->lines > 9999) {
 			r->lines = 9999;
 		}
diff -Naud 'gnucobol-svn/cobc/tree.h' 'gnucobol-svn-patched/cobc/tree.h'
Index: ./cobc/tree.h
--- ./cobc/tree.h	Sat Jun 27 22:18:11 2020
+++ ./cobc/tree.h	Tue Jul 21 08:14:00 2020
@@ -1532,6 +1532,7 @@
 	unsigned int		global:1;	/* IS GLOBAL declared */
 	unsigned int		has_declarative:1;/* Has Declaratives Code to be executed */
 	unsigned int		has_detail:1;	/* Has DETAIL line */
+	unsigned int            checked : 1;
 };
 
 #define CB_REPORT(x)	(CB_TREE_CAST (CB_TAG_REPORT, struct cb_report, x))
diff -Naud 'gnucobol-svn/libcob/mlio.c' 'gnucobol-svn-patched/libcob/mlio.c'
Index: ./libcob/mlio.c
--- ./libcob/mlio.c	Tue Jul 21 07:51:27 2020
+++ ./libcob/mlio.c	Tue Jul 21 07:53:39 2020
@@ -732,8 +732,8 @@
 		cob_free (name);
 	}
 	return status;
-#endif
 }
+#endif
 
 #endif
 
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Tue Jul 21 08:15:11 2020
# Generated by        : makepatch 2.03
# Recurse directories : Yes
# Excluded files      : (\A|/).*\~\Z
#                       (\A|/).*\.a\Z
#                       (\A|/).*\.bak\Z
#                       (\A|/).*\.BAK\Z
#                       (\A|/).*\.elc\Z
#                       (\A|/).*\.exe\Z
#                       (\A|/).*\.gz\Z
#                       (\A|/).*\.ln\Z
#                       (\A|/).*\.o\Z
#                       (\A|/).*\.obj\Z
#                       (\A|/).*\.olb\Z
#                       (\A|/).*\.old\Z
#                       (\A|/).*\.orig\Z
#                       (\A|/).*\.rej\Z
#                       (\A|/).*\.so\Z
#                       (\A|/).*\.Z\Z
#                       (\A|/)\.del\-.*\Z
#                       (\A|/)\.make\.state\Z
#                       (\A|/)\.nse_depinfo\Z
#                       (\A|/)core\Z
#                       (\A|/)tags\Z
#                       (\A|/)TAGS\Z
# p 'cobc/tree.c' 167727 1595340870 0100664
# p 'cobc/tree.h' 84549 1595340840 0100664
# p 'libcob/mlio.c' 23761 1595339619 0100664
#### End of ApplyPatch data ####

#### End of Patch kit [created: Tue Jul 21 08:15:11 2020] ####
#### Patch checksum: 86 2752 48598 ####
#### Checksum: 104 3445 41092 ####
